### ISSUE TITLE
worker: Clean up pointer juggling on dynamic args and args in RequestJob (server and client)

### ISSUE THREAD

**Issue by croissanne** [osbuild/osbuild-composer#1825]
blocked:
- [ ] deepmap>=1.8 in fedora (wait until fedora33 EOL)

this can be removed with deepmap>=1.8 and `x-oapi-codegen-extra-tags` and marking them as required but omitempty
```
	args := json.RawMessage{}
	if jr.Args != nil {
		args = *jr.Args
	}
	dynamicArgs := []json.RawMessage{}
	if jr.DynamicArgs != nil {
		dynamicArgs = *jr.DynamicArgs
	}
```

**Comment by ondrejbudai** [comment]
I think that pushing deepmap 1.8 into Fedora was basically impossible because of outdated deps so this will probably need to wait for Fedora 33 EOL: Tue 2021-11-16, see https://fedorapeople.org/groups/schedule/f-35/f-35-key-tasks.html

**Comment by Gundersanne** [comment]
Agreed, fine to wait until f33 EOL. We have some hacks in the spec file too which can be removed at that point.
